src/pages/catalog/Catalog.js

```diff
@@ -62,15 +106,23 @@ function Catalog() {
         };
         allCourses.push(course);
       }
-      localStorage.setItem(JSON.stringify(filter_key), JSON.stringify(allCourses));
+      localStorage.setItem(
+        JSON.stringify(filter_key),
+        JSON.stringify(filter_value.data)
+      );
       var cache_list = JSON.parse(localStorage.getItem("cache_list"));
       cache_list.push(filter_key);
       localStorage.setItem("cache_list", JSON.stringify(cache_list));
+      localStorage.setItem(
+        JSON.stringify(filter_key).concat("cards"),
+        JSON.stringify(allCourses)
+      );
```

Comment:
Might want to use `localStorage.removeItem(JSON.stringify(filter_key).concat("cards"))` at some point. Possibly use it with the `cache_list` in `Filter.js`

Possibly something like this at `line 37 Filter.js`
```
if (cache_list.length > 8) {
          var cache_card = cache_list[0]+'card';
          localStorage.removeItem(cache_card);
          localStorage.removeItem(cache_list[0]);
          cache_list = cache_list.shift();
}
```